source/adapters/native_cpu/enqueue.cpp

```diff
@@ -484,15 +508,17 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueMemBufferMap(
     ur_map_flags_t mapFlags, size_t offset, size_t size,
     uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
     ur_event_handle_t *phEvent, void **ppRetMap) {
-  std::ignore = hQueue;
   std::ignore = blockingMap;
   std::ignore = mapFlags;
   std::ignore = size;
-  std::ignore = numEventsInWaitList;
-  std::ignore = phEventWaitList;
-  std::ignore = phEvent;
 
+  urEventWait(numEventsInWaitList, phEventWaitList);
+  ur_event_handle_t event =
+      new ur_event_handle_t_(hQueue, UR_COMMAND_MEM_BUFFER_MAP);
+  event->tick_start();
   *ppRetMap = hBuffer->_mem + offset;
+  event->tick_end();
```

Comment:
This seems to just time fast pointer arithmetic (because it's just enqueue) which is probably less than the overhead of `std::lock_guard` in `tick_end`. Just creating the event like in `urEnqueueMemUnmap` below is probably fine.